sources/packages/backend/apps/api/src/route-controllers/assessment/assessment.controller.service.ts

```diff
@@ -320,32 +320,41 @@ export class AssessmentControllerService {
    * Get all pending and declined requests related to an application which would result
    * a new assessment when that request is approved.
    * @param applicationId application id.
-   * @param studentId student id.
+   * @param options options for request assessments,
+   * - `studentId` student id.
+   * - `showOfferingChange` will decide weather to show assessment
+   * request for offering change.
    * @returns assessment requests or exceptions for the student application.
    */
   async requestedStudentAssessmentSummary(
     applicationId: number,
-    studentId?: number,
+    options?: {
+      studentId?: number;
+      showOfferingChange?: boolean;
+    },
   ): Promise<RequestAssessmentSummaryAPIOutDTO[]> {
+    const showOfferingChange = options?.showOfferingChange ? true : false;
     const requestAssessmentSummary: RequestAssessmentSummaryAPIOutDTO[] = [];
-    const offeringChange =
-      await this.educationProgramOfferingService.getOfferingRequestsByApplicationId(
-        applicationId,
-        studentId,
-      );
-    if (offeringChange) {
-      requestAssessmentSummary.push({
-        id: offeringChange.id,
-        submittedDate: offeringChange.submittedDate,
-        status: offeringChange.offeringStatus,
-        requestType: RequestAssessmentTypeAPIOutDTO.OfferingRequest,
-        programId: offeringChange.educationProgram.id,
-      });
+    if (showOfferingChange) {
```

Comment:
we could have used `if(options?.showOfferingChange)` right, what is the benefit const at 336. ?